### PR TITLE
[ESS][8.18] Editing, exporting, and importing prebuilt rules

### PR DESCRIPTION
**Pull request by nastasha-solomon**
### Description
Partially addresses https://github.com/elastic/security-docs/issues/5061 by providing docs for editing, exporting, and importing prebuilt rules (customized and non-customized). Docs for updating customized prebuilt rules are at https://github.com/elastic/security-docs/pull/6568.

Twin 9.0 and Serverless PR: https://github.com/elastic/docs-content/pull/893

### Previews

- [Modify existing rules settings](https://security-docs_bk_6563.docs-preview.app.elstc.co/guide/en/security/8.x/rules-ui-management.html#edit-rules-settings): Made a few changes to this section:
  - Added requirements to the start of the section to explain subscription needs.
  - Updated instructions for editing single rules
- [Export and import rules](https://security-docs_bk_6563.docs-preview.app.elstc.co/guide/en/security/8.x/rules-ui-management.html#import-export-rules-ui): Made the following changes:
  - Added requirements to the start of the section to explain subscription needs for importing and exporting custom and prebuilt rules.  
  - Split up the instructions for importing and exporting rules to make the content more findable and readable
  - Added instructions for exporting individual rules

### CODE REVIEW

File: docs/detections/rules-ui-manage.asciidoc [hunk 1]
```diff
@@ -152,14 +154,15 @@ image::images/rule-snoozing.png[Rules snooze options,65%]
...
-You can export custom detection rules to an `.ndjson` file, which you can then import into another {elastic-sec} environment. 
-
-[NOTE]
-====
-You cannot export Elastic prebuilt rules, but you can duplicate a prebuilt rule, then export the duplicated rule.
+.Requirements
+[sidebar]
+--
+//Commenting out the first list item because need to confirm that the tables will be updated for 8.18 and 9.0.
+//* To use the import and export rules feature, you must have the appropriate subscription. Refer to the subscription page for https://www.elastic.co/subscriptions/cloud[Elastic Cloud] and {subscriptions}[Elastic Stack/self-managed] for the breakdown of available features and their associated subscription tiers.
```
Comment (nikitaindik) [inline docs/detections/rules-ui-manage.asciidoc:161]: Hey, Nastasha! 👋 I've just added a breakdown of which licenses/tiers are required for which import/export operations in [this docs ticket](https://github.com/elastic/security-docs/issues/5061). It's in the description under "Prebuilt rule export/import workflow". Please take a look, in case you want to expand on it here or add this info to the tables you mentioned.

File: docs/detections/rules-ui-manage.asciidoc [hunk 2]
```diff
@@ -58,43 +58,45 @@ For {ml} rules, an indicator icon (image:images/rules-table-error-icon.png[Error
 [[edit-rules-settings]]
 === Modify existing rules settings
 
-You can edit an existing rule's settings, and can bulk edit settings for multiple rules at once.
+You can edit an existing prebuilt or custom rule's settings, and can bulk edit settings for multiple rules at once.
```
Comment (nikitaindik) [inline docs/detections/rules-ui-manage.asciidoc:61]: I think we need to mention that it's only for users with Enterprise license / Security Complete tier. Added info into "Licensing restrictions" section at the bottom of ticket description [here](https://github.com/elastic/security-docs/issues/5061).

**Review by approksiu — COMMENTED** [review]
No comments outside of the need to mention licensing. Thanks Nastasha!

File: docs/detections/rules-ui-manage.asciidoc (hunk 1 above)
Comment (nastasha-solomon) [reply docs/detections/rules-ui-manage.asciidoc]: Modified this statement in [baf80d5](https://github.com/elastic/security-docs/pull/6563/commits/baf80d5ba461c9f082d449f068ea83f84d91f716).

File: docs/detections/rules-ui-manage.asciidoc (hunk 2 above)
Comment (nastasha-solomon) [reply docs/detections/rules-ui-manage.asciidoc]: Addressed in [baf80d5](https://github.com/elastic/security-docs/pull/6563/commits/baf80d5ba461c9f082d449f068ea83f84d91f716).

File: docs/detections/rules-ui-manage.asciidoc [hunk 3]
```diff
@@ -173,25 +176,40 @@ TIP: You can also use {kib}'s {kibana-ref}/managing-saved-objects.html#managing-
...
+.. Drag and drop the `.ndjson` file that contains the exported rules. 
+.. (Optional) Select the appropriate options to overwrite existing data:
+** *Overwrite existing detection rules with conflicting "rule_id"*: Updates existing rules if they match the `rule_id` value of any rules in the import file. Configuration data included with the rules, such as actions, is also overwritten.
+** *Overwrite existing exception lists with conflicting "list_id"*: Replaces existing exception lists with exception lists from the import file if they have a matching `list_id` value.
+** *Overwrite existing connectors with conflicting action "id"*: Updates existing connectors if they match the `action id` value of any rule actions in the import file. Configuration data included with the actions is also overwritten.
 .. Click *Import rule*.
-.. (Optional) If a connector is missing sensitive information after the import, a warning displays and you're prompted to fix the connector. In the warning, click *Go to connector*. On the Connectors page, find the connector that needs to be updated, click *Fix*, then add the necessary details.  
+
+The exported rules are imported to the Rules table.
```
Comment (approksiu) [inline docs/detections/rules-ui-manage.asciidoc:203]: Suggestuion: The imported rules are added to the Rules table.

File: docs/detections/rules-ui-manage.asciidoc [hunk 4]
```diff
@@ -152,14 +155,14 @@ image::images/rule-snoozing.png[Rules snooze options,65%]
...
 
-You can export custom detection rules to an `.ndjson` file, which you can then import into another {elastic-sec} environment. 
-
-[NOTE]
-====
-You cannot export Elastic prebuilt rules, but you can duplicate a prebuilt rule, then export the duplicated rule.
+.Requirements
+[sidebar]
+--
+* To learn which subscription you need for exporting and importing custom rules and prebuilt rules (modified and unmodified), refer to the https://www.elastic.co/subscriptions/cloud[Elastic Cloud] subscription page.
```
Comment (approksiu) [inline docs/detections/rules-ui-manage.asciidoc:161]: There is also Elastic Stack subscriptions, should we mention it too? https://www.elastic.co/subscriptions

**Review by nikitaindik — APPROVED** [review]
Thank you, @nastasha-solomon! After adding a mention of licensing restrictions this PR LGTM now.

File: docs/detections/rules-ui-manage.asciidoc (hunk 3 above)
Comment (nastasha-solomon) [reply docs/detections/rules-ui-manage.asciidoc:203]:
```suggestion
The imported rules are added to the Rules table. 
```

File: docs/detections/rules-ui-manage.asciidoc (hunk 4 above)
Comment (nastasha-solomon) [reply docs/detections/rules-ui-manage.asciidoc:161]: Yeah, good idea to ref the self-managed subscriptions page. Thanks!

```suggestion
* To learn which subscription you need for exporting and importing custom rules and prebuilt rules (modified and unmodified), refer to the subscription page for https://www.elastic.co/subscriptions/cloud[{ecloud}] and {subscriptions}[{stack}/self-managed]
```

File: docs/detections/rules-ui-manage.asciidoc [hunk 5]
```diff
@@ -152,14 +155,14 @@ image::images/rule-snoozing.png[Rules snooze options,65%]
...
+.Requirements
+[sidebar]
+--
+* To learn which subscription you need for exporting and importing custom rules and prebuilt rules (modified and unmodified), refer to the subscription page for https://www.elastic.co/subscriptions/cloud[{ecloud}] and {subscriptions}[{stack}/self-managed]
+* At minimum, your role needs `Read` privileges for the **Action and Connectors** feature to import rules with actions. To overwrite or add new connectors, you need `All` privileges. Refer to <<enable-detections-ui>> to learn more about the required privileges for managing rules.
+--
 
-If you try to export with both prebuilt and custom rules selected, only the custom rules are exported.
-====
+You can export prebuilt rules and custom rules to an `.ndjson` file, which you can then (with the proper subscription) import into another {elastic-sec} environment.
```
Comment (nastasha-solomon) [inline docs/detections/rules-ui-manage.asciidoc:165]: @nikitaindik just want to verify that the second part of statement (abut importing rules) is still ok. I added the caveat to account for the use case of importing customized rules. But maybe it's no longer needed with the changes being made in https://github.com/elastic/kibana/pull/214577?

Comment (nikitaindik) [reply docs/detections/rules-ui-manage.asciidoc:165]: You're right. With recent changes in PR #241577, users on Basic/Platinum/Security Essentials will be able to import prebuilt rules (both modified and non-modified). They won't be able to customize prebuilt rules though, so upgrade flyout UI will be readonly for them.

Comment (nastasha-solomon) [reply docs/detections/rules-ui-manage.asciidoc:165]: I'll remove the caveat in the parenthesis then. Thanks!

```suggestion
You can export prebuilt rules and custom rules to an `.ndjson` file, which you can then import into another {elastic-sec} environment. 
```

**Review by approksiu — APPROVED** [review]
LGTM! Thanks @nastasha-solomon !

File: docs/detections/rules-ui-manage.asciidoc [hunk 6]
```diff
@@ -173,25 +176,40 @@ TIP: You can also use {kib}'s {kibana-ref}/managing-saved-objects.html#managing-
...
+
+The imported rules are added to the Rules table. 
+
+[NOTE] 
+=====
+If the prebuilt rule package doesn't have the original version of a rule that you're importing, the rule is marked as `Modified` when both the following criteria are met:
+
+- The rule's ID (`rule_id`) is identical to that of an already installed prebuilt rule.
+- The imported rule's settings are different from the currently installed rule. 
+=====
```
Comment (banderror) [inline docs/detections/rules-ui-manage.asciidoc:211]: There are many more edge cases besides this one that we have to handle. We're currently trying to cover them all in this [PR](https://github.com/elastic/kibana/pull/213434) in a test plan under the "Handling missing base versions" and other sections. I don't think we should attempt to describe any of this under-the-hood complexity in the user-facing docs. The user shouldn't have to deal with this complexity and shouldn't even know that there's a package with prebuilt rules. The package is the implementation details of the prebuilt rules delivery mechanism.

I'd suggest to delete this note from the docs.

File: docs/detections/rules-ui-manage.asciidoc [hunk 7]
```diff
@@ -58,43 +58,46 @@ For {ml} rules, an indicator icon (image:images/rules-table-error-icon.png[Error
 [[edit-rules-settings]]
 === Modify existing rules settings
 
-You can edit an existing rule's settings, and can bulk edit settings for multiple rules at once.
+You can edit an existing custom rule's settings and can bulk edit settings for multiple rules at once. With the https://www.elastic.co/subscriptions/cloud[Enterprise] subscription, you can edit Elastic prebuilt rules and bulk modify them too.
```
Comment (banderror) [inline docs/detections/rules-ui-manage.asciidoc:61]: Let's mention that editing rule actions (notification and response actions) for prebuilt rules is allowed on any subscription.

**Review by banderror — COMMENTED** [review]
@nastasha-solomon Thank you, a few suggestions and comments that I think we should address in this PR.

File: docs/detections/rules-ui-manage.asciidoc (hunk 7 above)
Comment (nastasha-solomon) [reply docs/detections/rules-ui-manage.asciidoc]: Added in [bb2e2cc](https://github.com/elastic/security-docs/pull/6563/commits/bb2e2cc0cb8a081dad416e0f58e86bb885234439).

File: docs/detections/rules-ui-manage.asciidoc (hunk 6 above)
Comment (nastasha-solomon) [reply docs/detections/rules-ui-manage.asciidoc]: Sounds good - I debated about adding this note, but am fine to remove it. 

Removed in [bb2e2cc](https://github.com/elastic/security-docs/pull/6563/commits/bb2e2cc0cb8a081dad416e0f58e86bb885234439).

File: docs/detections/rules-ui-manage.asciidoc [hunk 8]
```diff
@@ -152,14 +155,14 @@ image::images/rule-snoozing.png[Rules snooze options,65%]
...
 
-You can export custom detection rules to an `.ndjson` file, which you can then import into another {elastic-sec} environment. 
-
-[NOTE]
-====
-You cannot export Elastic prebuilt rules, but you can duplicate a prebuilt rule, then export the duplicated rule.
+.Requirements
+[sidebar]
+--
+* To learn which subscription you need for exporting and importing custom rules and prebuilt rules (modified and unmodified), refer to the subscription page for https://www.elastic.co/subscriptions/cloud[{ecloud}] and {subscriptions}[{stack}/self-managed]
```
[the remark this answers is not in the document]
Comment (nastasha-solomon) [reply docs/detections/rules-ui-manage.asciidoc]: Addressed in [bb2e2cc](https://github.com/elastic/security-docs/pull/6563/commits/bb2e2cc0cb8a081dad416e0f58e86bb885234439).

File: docs/detections/rules-ui-manage.asciidoc [hunk 9]
```diff
@@ -58,43 +58,46 @@ For {ml} rules, an indicator icon (image:images/rules-table-error-icon.png[Error
...
 +
 NOTE: Rule actions won't run during a {kibana-ref}/maintenance-windows.html[maintenance window]. They'll resume running after the maintenance window ends.
-
++
 ** *Update rule schedules*: Update the <<rule-schedule,schedules>> and look-back times on all selected rules.
 ** *Apply Timeline template*: Apply a specified <<timeline-templates-ui, Timeline template>> to the selected rules. You can also choose *None* to remove Timeline templates from the selected rules.
+** *Export*: Export the selected rules to an `.ndjson` file also includes any actions, connectors, and exception lists related to the exported rules.
+** *Manual run*: Manually run the specified rules for a specified period of time. This option is only available for enabled rules.  
+** *Disable*: Turn the selected rules off.
+** *Delete*: Remove the selected rules.
```
[the remark this answers is not in the document]
Comment (nastasha-solomon) [reply docs/detections/rules-ui-manage.asciidoc]: Lot's of great feedback here. For now, I've removed the non-editing actions to keep the section focused on bulk actions that actually modify the rules' settings. I've also created https://github.com/elastic/security-docs-internal/issues/63 to track your suggestions for additional changes so I don't forget about them.

[the remark this answers is not in the document]
Comment (nastasha-solomon) [reply docs/detections/rules-ui-manage.asciidoc]: Comment partially addressed in [46d9559](https://github.com/elastic/security-docs/pull/6563/commits/46d9559ad5cca37583e90aa7b678069f11dedede).

File: docs/detections/rules-ui-manage.asciidoc [hunk 10]
```diff
@@ -173,25 +173,31 @@ TIP: You can also use {kib}'s {kibana-ref}/managing-saved-objects.html#managing-
...
 
 . Find *Detection rules (SIEM)* in the navigation menu or by using the {kibana-ref}/introduction.html#kibana-navigation-search[global search field]. 
-. To export rules:
-.. In the Rules table, select the rules you want to export.
-.. Select *Bulk actions* -> *Export*, then save the exported file.
-. To import rules:
-+
-NOTE: To import rules with actions, you need at least `Read` privileges for the `Action and Connectors` feature. To overwrite or add new connectors, you need `All` privileges for the `Actions and Connectors` feature. To import rules without actions,  you don't need `Actions and Connectors` privileges. Refer to <<enable-detections-ui>> for more information.
+. Do one of the following:
+** Export a single rule: Find the rule in the Rules table, then select  **All actions** ->**Export**. Alternatively, export the rule from its details page (click on the rule name to open its details, then click **All actions** ->**Export**).
```
Comment (nastasha-solomon) [inline docs/detections/rules-ui-manage.asciidoc:182]:
```suggestion
** Export a single rule: Find the rule in the Rules table, then select **All actions** ->**Export**. Alternatively, export the rule from its details page (click on the rule name to open its details, then click **All actions** ->**Export**).
```

File: docs/detections/rules-ui-manage.asciidoc [hunk 11]
```diff
@@ -173,25 +173,31 @@ TIP: You can also use {kib}'s {kibana-ref}/managing-saved-objects.html#managing-
...
 
 . Find *Detection rules (SIEM)* in the navigation menu or by using the {kibana-ref}/introduction.html#kibana-navigation-search[global search field]. 
-. To export rules:
-.. In the Rules table, select the rules you want to export.
-.. Select *Bulk actions* -> *Export*, then save the exported file.
-. To import rules:
-+
-NOTE: To import rules with actions, you need at least `Read` privileges for the `Action and Connectors` feature. To overwrite or add new connectors, you need `All` privileges for the `Actions and Connectors` feature. To import rules without actions,  you don't need `Actions and Connectors` privileges. Refer to <<enable-detections-ui>> for more information.
+. Do one of the following:
+** Export a single rule: Find the rule in the Rules table, then select **All actions** ->**Export**. Alternatively, export the rule from its details page (click on the rule name to open its details, then click **All actions** ->**Export**).
```
Comment (nastasha-solomon) [inline docs/detections/rules-ui-manage.asciidoc:182]:
```suggestion
** Export a single rule: Find the rule in the Rules table, then select **All actions** -> **Export**. Alternatively, export the rule from its details page (click on the rule name to open its details, then click **All actions** -> **Export**).
```

**Review by banderror — APPROVED** [review]
Thank you @nastasha-solomon, LGTM 👍

**Review by natasha-moore-elastic — APPROVED** [review]
Nice work!
